---
title: "R Notebook"
output: html_notebook
---

We're going to use the [curatedMetagenomicsData](<https://waldronlab.io/curatedMetagenomicData/>) library to get our reference sources.\
This library contains curated metadata for human metagenomic samples, as well as pre-preprocessed taxonomic profiles using MetaPhlan 3.0

```{r}
library(curatedMetagenomicData)
library(tidyverse)
library(mia)
```

```{r}
sampleMetadata %>% 
  filter(body_site=='stool' & antibiotics_current_use  == 'no' & disease == 'healthy') %>% 
  group_by(non_westernized) %>% 
  sample_n(100) %>% 
  ungroup() -> selected_samples
```

```{r}
selected_samples %>% 
  returnSamples("relative_abundance") -> rel_ab
```

```{r}
data_ranks = splitByRanks(rel_ab)
```

```{r}
for (r in names(data_ranks)){
  print(r)
  assay_rank = as.data.frame(assay(data_ranks[[r]]))
  print(paste0("../../data/curated_metagenomics/modern_sources_",tolower(r),".csv"))
  write.csv(assay_rank, paste0("../../data/curated_metagenomics/modern_sources_",tolower(r),".csv"))
}
```


```{r}
selected_samples %>% 
  as_tibble() %>% 
  write_csv("../../data/metadata/curated_metagenomics_modern_sources.csv")
```
